src/io.jl

```diff
@@ -74,6 +75,75 @@ hg_save(
     open(io -> hg_save(io, h, format), fname, "w")
 
 
+"""
+    hg_save(io::IO, h::Hypergraph, format::HIF_Format)
+
+Saves a hypergraph `h` to an output stream `io` in `HIF` format.
+
+If `h` has `Composite Types` either for vertex metadata or hyperedges metadata,
+the user has to explicit tell the JSON3 package about it, for instance using:
+
+`JSON3.StructType(::Type{MyType}) = JSON3.Struct()`.
+
+See the (JSON3.jl documentation)[https://github.com/quinnj/JSON3.jl] for more details.
+
+"""
+function hg_save(io::IO, h::Hypergraph, format::HIF_Format)
+    _ = format
+
+    json_hg = Dict{Symbol, Any}()
+    incidences = []
+    v_meta = h.v_meta
+    he_meta = h.he_meta
+
+    if any(isnothing, h.v_meta)
+        v_meta = [i for i in 1:length(h.v_meta)]
+    end
+
+    if any(isnothing, h.he_meta)
+        he_meta = [i for i in 1:length(h.he_meta)]
+    end
+
+    node_dict = Dict(i => val for (i, val) in pairs(v_meta))
+    edge_dict = Dict(i => val for (i, val) in pairs(he_meta))
+
+
+    types = collect(typeof(h).parameters)
+    V = types[2]
+    E = types[3]
+
+    for node_idx in 1:length(v_meta)
+        for edge_idx in 1:length(he_meta)
```

Comment:
use `eachindex` as this is the recommended syntax
https://discourse.julialang.org/t/best-practices/85138